geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/function/CurrentDateFunction.scala

```diff
@@ -8,16 +8,33 @@
 
 package org.locationtech.geomesa.filter.function
 
-import java.time.{ZoneOffset, ZonedDateTime}
+import java.time.{Duration, ZoneOffset, ZonedDateTime}
 import java.util.Date
 
+import org.geotools.data.Parameter
 import org.geotools.filter.FunctionExpressionImpl
 import org.geotools.filter.capability.FunctionNameImpl
 
 class CurrentDateFunction extends FunctionExpressionImpl(CurrentDateFunction.Name) {
-  override def evaluate(o: AnyRef): AnyRef = Date.from(ZonedDateTime.now(ZoneOffset.UTC).toInstant)
+  private var offset = ""
+
+  override def evaluate(o: AnyRef): AnyRef = {
+    if (offset.isEmpty && getParameters.size > 0) {
+      offset = getExpression(0).evaluate(null).asInstanceOf[String]
+    }
+
+    val now = ZonedDateTime.now(ZoneOffset.UTC).toInstant
+
+    if (offset.isEmpty) {
+      Date.from(now)
+    } else {
+      Date.from(now.minus(Duration.parse(offset)))
```

Comment:
it might make sense to use `plus` here instead of `minus` - that way you could do `currentDate('P1D')` for future dates, and `currentDate('-P1D')` for past dates. otherwise it would be the inverse, which makes some sense if you never consider future dates but seems a bit confusing when you do